### ISSUE TITLE
docker-compose settings do not persist database by default

### ISSUE THREAD

**Issue by fwenzel** [tootsuite/mastodon#6716]
I can see no reason why the database would be considered volatile data by default, so that when you run `docker-compose down` you lose your entire mastodon instance's data.

I have a nightly backup and will restore it now, but this is a time bomb by default.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by Gargron** [comment]
There are a lot of warnings about this in docker-compose.yml and the documentation. I would've made it persistent by default if I was starting from scratch, but because it's already this way, changing it might break existing installations of people who haven't been paying attention... There's risk of data loss either way.

**Comment by fwenzel** [comment]
That's fair. Is there a way out of it?

**Comment by fwenzel** [comment]
Specifically, I am wondering if there should be a migration path to persist the data after the fact and un-break the situation. It seems like we agree status quo is a terrible outcome

**Comment by Gargron** [comment]
Right. So in reality, the data is *always* persisted. Except, unless you give it a host path, it gives it a random host path. Using docker inspect you can find where that path is, and, given that postgres/redis are turned off, copy over the contents to another directory (or, just use that path in docker-compose.yml explicitly, I guess, I haven't tried that so YMMV)

**Comment by fwenzel** [comment]
Thank you

**Comment by algernon** [comment]
What I did was to figure out the mounted directory (via `docker inspect mastodon_db_1`, etc), stop (but not remove!) the container, move the data to `./postgresql` (and to `./redis`), uncomment the lines in `docker-compose.yml`, ran `docker-compose down`, and then did the usual upgrade steps.

This seems to have worked, and because I moved the data, I didn't even need extra disk space for a copy. (which would have been problematic, because my VM doesn't have enough space for that)

**Comment by fwenzel** [comment]
This is great, thanks @algernon
